packages/web3-core-method/src/index.js

```diff
@@ -738,9 +742,16 @@ Method.prototype.buildCall = function () {
                 }
             }
 
+
             return method.requestManager.send(payload, sendTxCallback);
         };
 
+        if (isSendTx) {
+            setTimeout(() => {
+                defer.eventEmitter.emit('sending');
+            }, 10);
+        }
```

Comment:
WDYT about the idea in the initial design that this always be fired after all the preparatory calls (like getGasPrice below here?) 

> Fired immediately before transmitting the transaction request - connection to provider must be open when this is fired. Intermediate requests such as eth_chainId, eth_estimateGas, or eth_gasPrice must be completed prior this event being fired.

Wonder if it might be useful to pass the payload as an argument to the handler for both `sending` and `sent`. Another idea floated in #3550 is that this event could help w/ debugging.

Reply:
sending payload with handler sounds like a nice idea, I will add that.

ah right sure I’ll have to add it in a few places then to satisfy branching conditional logic specific to the flow but that’s no problem.

Reply:
both resolved in https://github.com/ethereum/web3.js/pull/3550/commits/34bdd39aad540ccf03bd22f812ed2ec8584291b0 👍